lib/std/math/exp.zig

```diff
@@ -36,7 +43,7 @@ fn exp32(x_: f32) f32 {
     hx &= 0x7FFFFFFF;
 
     if (math.isNan(x)) {
-        return x;
+        return math.nan(f32);
```

Comment:
This results in slightly different behavior in that the nan bit-representation returned will not necessarily be the same as the provided nan since there are many possible bit-representations. Probably won't cause an issue in practice but probably not worth changing to avoid surprises.

it's interesting to note the following the C standard

_Recommended practice
13 If a function with one or more NaN arguments returns a NaN result, the result should be the same
as one of the NaN arguments (after possible type conversion), except perhaps for the sign_

Reply:
Yes, you're right, I should revert this change. After opening this PR I performed the property-based testing against GCC and found that this implementation disagrees with GCC (and Musl) on the NaN handling. The reason I changed it is I think the Zig handling was already inconsistent, so I was just trying to make it consistent! I'll take another pass over this PR sometime hopefully soon.

Reply:
This will be resolved by https://github.com/ziglang/zig/pull/10415